### PR TITLE
Use custom module map for the framework + Add podspec (#3219)

### PR DESCRIPTION
**Pull request by andersio**
Merged in also @ikesyo's #3219 (official podspec).

-
This is a change for #3219, since CocoaPods doesn't seem to support `SWIFT_INCLUDE_DIR`. This approach also seems to hide the private headers without problem, too.

### CODE REVIEW

**Comment by andersio** [comment]
Seems running fine. ;)
https://github.com/ReactiveCocoa/ReactiveCocoa/pull/3210#issuecomment-255038660

**Comment by dylanmaryk** [comment]
Exactly what I'm looking for! Once someone merges this, can "master" please be merged into "add-podspec" so those of us using CocoaPods can utilise this change?

**Comment by ikesyo** [comment]
Thanks @andersio, I've slightly updated the podspec in this branch.

File: ReactiveCocoa/module.modulemap [hunk 1]
```diff
@@ -0,0 +1,7 @@
+framework module ReactiveCocoa {
```
Comment (NachoSoto) [inline ReactiveCocoa/module.modulemap]: :+1:

**Pull request merged**